### PR TITLE
WT-12375 Elaborate on the prepare state comment in btmem.h

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

**Review by agorrod — APPROVED** [review]
LGTM - thanks Luke, it's a big improvement.

File: src/include/btmem.h [hunk 1]
```diff
@@ -791,36 +791,123 @@ struct __wt_page {
...
+ * In this scenario it is possible the writer thread context switches between 3 and 4, and therefore
+ * the reader may see a timestamp of 10 for start_ts and wrongly attribute it to the prepare
+ * timestamp. This creates a need for the WT_PREPARE_LOCKED state. The writer thread now performs:
+ *  1: start_ts = 5
+ *  2: prepare_state = WT_PREPARE_INPROGRESS
+ *  3: prepare_state = WT_PREPARE_LOCKED
+ *  3: start_ts = 10
+ *  4: prepare_state = WT_PREPARE_RESOLVED
+ *
+ * The reader thread can in this scenario read any prepare state and behave correctly, if it reads
```
Comment (luke-pearson) [inline src/include/btmem.h:878]: need comma between "can" and "in"

**Review by ajmorton — APPROVED** [review]
LGTM 👍

**Comment by luke-pearson** [comment]
evergreen merge

**Pull request merged**